lib/dry/system/provider_sources/settings/config.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "dry/configurable"
+require_relative "file_loader"
+
+module Dry
+  module System
+    module ProviderSources
+      module Settings
+        InvalidSettingsError = Class.new(ArgumentError) do
+          # @api private
+          def initialize(errors)
+            message = <<~STR
+              Could not load settings. The following settings were invalid:
+
+              #{setting_errors(errors).join("\n")}
+            STR
+
+            super(message)
+          end
+
+          private
+
+          def setting_errors(errors)
+            errors.sort_by { |k, _| k }.map { |key, error| "#{key}: #{error}" }
+          end
+        end
+
+        class Config
+          class << self
+            def load(root, env)
+              env_data = load_files(root, env)
+
+              errors = {}
```

Comment:
you can move it to the `tap`'s block I think, or use `errors = ...each_with_object({}) do |setting_name, errors|`

Reply:
Ah yes, thanks! It was in that outer scope still as a hangover from the previous implementation, and I didn't notice the chance to move it in.

I think I might leave the loop structure as it is currently, because populating the errors hash is somewhat of a secondary effect (the primary effect is to assign the settings), and if we changed it to `errors = ...each_with_object({}) do |setting_name, errors|` it might look like the main purpose of the loop is to populate the errors only.